src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/SqsMessageReceiverIntTest.java

```diff
@@ -258,6 +266,52 @@ public void givenNewCase_whenExactPersonRecordFound_thenSetPersonIdOnDefendant()
+        featureFlags.setFlagValue("save_person_id_to_court_case_service", true);
+        var s3Key = "test-key";
+
+        s3AsyncClient.putObject(builder -> builder.bucket(s3Bucket).key(s3Key).build(), Paths.get(BASE_PATH + "/common-platform/hearing-with-legal-entity-defendant.json"));
+
+        String messageBody = "[ \"software.amazon.payloadoffloading.PayloadS3Pointer\", {\n" +
+            String.format("  \"s3BucketName\" : \"%s\",\n", s3Bucket) +
+            String.format("  \"s3Key\" : \"%s\"\n", s3Key) +
+            "} ]";
+
+        publishMessage(messageBody, Map.of("messageType",
+            MessageAttributeValue.builder().dataType("String").stringValue("COMMON_PLATFORM_HEARING").build(),
+            "hearingEventType",
+            MessageAttributeValue.builder().dataType("String").stringValue("ConfirmedOrUpdated").build(),
+            "ExtendedPayloadSize",
+            MessageAttributeValue.builder().dataType("String").stringValue("268444").build()
+        ));
+        await()
+            .atMost(10, TimeUnit.SECONDS)
+            .until(() -> countPutRequestsTo("/hearing/E10E3EF3-8637-40E3-BDED-8ED104A380AC") == 1);
+
+        MOCK_SERVER.verify(
+            putRequestedFor(urlMatching("/hearing/E10E3EF3-8637-40E3-BDED-8ED104A380AC"))
+                .withRequestBody(matchingJsonPath("caseId", equalTo("D2B61C8A-0684-4764-B401-F0A788BC7CCF")))
+                .withRequestBody(matchingJsonPath("hearingId", equalTo("E10E3EF3-8637-40E3-BDED-8ED104A380AC")))
+                .withRequestBody(matchingJsonPath("caseNo", equalTo("D2B61C8A-0684-4764-B401-F0A788BC7CCF")))
+                .withRequestBody(matchingJsonPath("hearingDays[0].courtRoom", equalTo("Crown Court 3-1")))
+                .withRequestBody(matchingJsonPath("defendants[0].type", equalTo("PERSON")))
+                .withRequestBody(matchingJsonPath("defendants[0].defendantId", equalTo("0ab7c3e5-eb4c-4e3f-b9e6-b9e78d3ea199")))
+                .withRequestBody(matchingJsonPath("defendants[0].personId", equalTo("e374e376-e2a3-11ed-b5ea-0242ac120002")))
+
+                .withRequestBody(matchingJsonPath("defendants[0].crn", equalTo("X346204")))
+                .withRequestBody(matchingJsonPath("defendants[1].type", equalTo("ORGANISATION")))
+                .withRequestBody(matchingJsonPath("defendants[1].defendantId", equalTo("903c4c54-f667-4770-8fdf-1adbb5957c25")))
+                .withRequestBody(matchingJsonPath("defendants[1].personId", equalTo("e374e376-e2a3-11ed-b5ea-0242ac120002")))
+        );
+
+
+        verify(telemetryService).trackHearingMessageReceivedEvent(any(String.class));
+        verify(telemetryService).trackNewHearingEvent(any(Hearing.class), any(String.class));
+        verify(telemetryService).trackOffenderMatchEvent(any(Defendant.class), any(Hearing.class), any(MatchResponse.class));
+        verifyNoMoreInteractions(telemetryService);
+    }
+
```

Comment:
🎩

Reply:
Never seen the tophat before, is that like a thumbs up? 😅 or something needs to change?

Reply:
I like it, a neat test. Might be even better if the json for the message was stored in a separate file so we could copy and paste it for the other consumer projects :-)

Reply:
awesome. I've added a large json in [court-hearing-event-receiver here](https://github.com/ministryofjustice/court-hearing-event-receiver/blob/73990330d3a34c0d9d86f30857d3427f33464e00/src/test/resources/json/large-hearing-update.json)

As for this test, i uploaded a small hearing.json to s3 and just arranged an sns message with an s3pointer.

Reply:
👍 the message at the top of this PR is what defines the contract between CHER and the three downstream dependencies - that is what we will use in our tests and no doubt CEMO will do the same